attest/api/src/convert.rs

```diff
@@ -181,12 +182,8 @@ impl From<EnclaveMessage<NonceSession>> for NonceMessage {
 ///     let rust_type = TYPENAME::try_from(prost)?;
 /// ```
 pub(crate) fn encode_to_protobuf_vec<T: ProtoMessage>(msg: &T) -> Result<Vec<u8>, ConversionError> {
-    let mut bytes = vec![];
-    let mut stream = CodedOutputStream::vec(&mut bytes);
-    msg.write_to_with_cached_sizes(&mut stream)
-        .map_err(|e| ConversionError::Other(e.to_string()))?;
-    stream
-        .flush()
+    let bytes = msg
+        .write_to_bytes()
```

Comment:
Note this change. I missed the `write_to_bytes()` function originally and the `write_to_with_cached_sizes()` specifically calls out
> Sizes of this messages and nested messages must be cached by calling `compute_size` prior to this call.

https://github.com/stepancheg/rust-protobuf/blob/f7259280e1b64a8bafd505b688d35f93552d0a38/protobuf/src/message.rs#L40

While wrapping this one function call `write_to_bytes()` with `encode_to_protobuf_vec()` is a bit questionable normally, I think since alot of the Message API is C like with temporally dependent functions keeping this wrapped is probably best.